pyomo/contrib/mindtpy/algorithm_base_class.py

```diff
@@ -1626,7 +1684,11 @@ def solve_main(self):
                         "No-good cuts are added and GOA algorithm doesn't converge within the time limit. "
                         'No integer solution is found, so the CPLEX solver will report an error status. '
                     )
-            return None, None
+            # Value error will be raised if the MIP problem is unbounded and appsi solver is used when loading solutions. Although the problem is unbounded, a valid result is provided and we do not return None to let the algorithm continue.
+            if 'main_mip_results' in locals():
+                return self.mip, main_mip_results
```

Comment:
I tried. However, it's not easy to find a problem to generate a model with an unbounded MIP problem, with the initialization strategy. Tbh, I forgot which instance resulted in this situation in the benchmark tests.

Reply:
You could always make the whole problem unbounded--then you would be sure to hit this (as long as you didn't discover it was unbounded during initialization I guess). I think there are a couple small examples that might be useful to build off of in `test_gdpopt.py`. Another option would be to just directly call the `solve_main` method and skip initialization.

Reply:
This code will not be triggered for all unbounded problems. It only happens when the problem is unbounded and the heuristic method in the solver also finds a feasible solution, **as follows**. That's why the problem is unbounded and still calls the `load_solutions` function. I can manually reproduce this situation by changing the code of MindtPy.  However, I didn't find an instance that could result in this situation without any change. It might not be worth to introduce some redundant code just to trigger this error. I will try to increase the code coverage in other parts.
```
Gurobi Optimizer version 10.0.2 build v10.0.2rc0 (mac64[x86])

CPU model: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
Thread count: 6 physical cores, 12 logical processors, using up to 12 threads

Optimize a model with 4 rows, 6 columns and 8 nonzeros
Model fingerprint: 0xb497cb3e
Variable types: 5 continuous, 1 integer (0 binary)
Coefficient statistics:
  Matrix range     [1e+00, 4e+00]
  Objective range  [1e+00, 1e+00]
  Bounds range     [0e+00, 0e+00]
  RHS range        [3e+00, 2e+01]
Found heuristic solution: objective -3.60103e+09
Presolve removed 3 rows and 3 columns
Presolve time: 0.00s

Explored 0 nodes (0 simplex iterations) in 0.00 seconds (0.00 work units)
Thread count was 1 (of 12 available processors)

Solution count 1: -3.60103e+09 
No other solutions better than 0

Model is unbounded
Best objective -3.601030018124e+09, best bound -, gap -
```